src/plugins/application_config/server/types.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IScopedClusterClient } from 'src/core/server';
+
+export interface ApplicationConfigPluginSetup {
+  getConfigurationClient: (inputOpenSearchClient: IScopedClusterClient) => ConfigurationClient;
```

Comment:
I think this should be part of ApplicationConfigPluginStart interface.

Reply:
Answered here https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5855/files#r1505186060